typescript/cli/src/deploy/core.ts

```diff
@@ -161,6 +168,10 @@ async function runIsmStep(
   const isIsm = isZODISMConfig(ismConfigPath);
   // separate flow for 'ism' and 'ism-advanced' options
   if (isIsm) {
```

Comment:
Can you confirm this check will only apply for advanced configs? The name makes me think it's for any ism config

Reply:
yes, this is just for advancedIsmConfig.

If I recall correctly, a certain two reviewers didn't like the idea of calling it `isAdvancedIsm` and `readAdvancedIsmConfig` so I changed it to isIsm, saying that this will cause confusion further down the road. 👀

Reply:
IIRC we didn't like the name `isZODISMConfig` because of the Zod part, zod being an implementation detail and not a property of the config itself. I still think that :)

My confusion here is because of the variable name though (`isISM`)

I'm in favor of renaming `isZODISMConfig` to `isAdvancedIsm` if that's what you prefer!

Reply:
made it isAdvancedIsm